[FEAT] 섹션 생성 API 구현

JSeHoone opened the pull request:
## ❗️ 이슈 번호
#11 

## 📝 작업 내용
**1. SectionService 인터페이스, SectionServiceImpl 구현체**
- 구현체 내부에 registerSection 메서드를 정의하여 비즈니스 로직을 만들었습니다.
- **Course가 없는 경우**, Section에 **동일한 course_id, order_no를 가질 경우** throw를 반환하도록 하였습니다.
- Controller로 넘기기 전에 **SectionCreateResponseDTO로 변환**하여 return했습니다.

**2. ResponseDTO, RequestDTO**
- **Validation 어노테이션을 사용**하여 필수 요청, 응답 데이터를 관리했습니다.
- 응답 시 `@Getter`어노테이션을 사용하여 **직렬화 문제를 해결**하였습니다.

**3 . SectionController**
- SectionService에서 응답 받은 값을 **BaseResponse의 Data에 넣어 응답**하였습니다
- Try Catch 구문을 사용하여 **Service에서 발생한 throw를 받아 응답**하도록 구현하였습니다.

**4. SectionRepository**
- Section생성 시 동일한 **course_id와 order_no를 가지는 값이 있는지 확인**하는 findByCourseIdAndOrderNo를 정의하였습니다.

## 💭 주의 사항

## 💡 리뷰 포인트
1. SectionService에서 IllegalArgumentException, IllegalStateException으로 나눠서 오류 처리를 구분하였는데 다른 방법이 있을까요 ?
2. controller에서 CREATE(201), NOT_FOUND(404), BAD_REQUEST(400)으로 구분했는데 postman에서는 동일하게 200으로 반환되는 것 같은데 상관 없을까요 ? (응답 메시지의 'status' 필드는 201, 404, 400으로 표기되어있기는 합니다 !

<img width="1004" height="329" alt="image" src="https://github.com/user-attachments/assets/c173226b-f070-4d65-9779-5e87c9ba9c8b" />
<img width="1007" height="329" alt="image" src="https://github.com/user-attachments/assets/a35e878a-b844-4f9b-9c0b-087e1097d2f0" />
<img width="1008" height="329" alt="image" src="https://github.com/user-attachments/assets/8de6ce52-3722-4cae-ab8c-eb856952fa8c" />

JSeHoone committed:
feat : Factory Method 추가

- Section 객체를 생성할 때, Factory Method를 이용해서 생성하기

JSeHoone committed:
feat : findByCourseIdAndOrderNo 메서드 추가

- Section내 동일한 CourseId와 OrderNo가 있는지 체크하기 위해서 만들었습니다.

JSeHoone committed:
feat : SectionController - createSection

- Section을 생성하는 API입니다.
- REST URL대로 만들기 위해 노력했습니다.
- Service내 Throw가 발생했을 때, Controller에서 처리합니다.

JSeHoone committed:
feat : Section 생성 시 요청 DTO

- Section 생성 요청이 들어올 때 사용하는 DTO입니다.
- Validation 라이브러리를 사용하여 필수 값이 없으면 에러 응답이 반환되도록 만들었습니다.
- title은 빈 값이 올 수 없도록 설정하였습니다.
- DTO 생성자는 public으로 만들었습니다.

JSeHoone committed:
feat : Section 생성 시 응답 DTO

- Section 생성 후 응답 시 data 필드에 들어가는 값에 대한 DTO입니다.
- Controller에서 최종 응답 시 Data에 직렬화가 되지 않는 이슈를 해결하기 위해 `Getter`어노테이션을 넣어 해결했습니다.
- Validation 라이브러리를 사용하여 필수 값을 받지 않으면 오류가 발생하도록 정의하였습니다.
- title은 빈 값이 올 수 없도록 정의하였습니다.
- 생성자는 public으로 정의하였습니다.

JSeHoone committed:
feat : SectionService interface

- SectionService의 인터페이스를 정의하였습니다.

JSeHoone committed:
feat : SectionServiceImpl

- 인터페이스로 정의한 SectionService를 상속받은 구현체입니다.
- `Service`어노테이션을 사용하여 Bean Context에 등록하였습니다.
- 생성자에 `@Autowired`를 사용하여 의존성을 주입하였습니다.
- registerSection 메서드를 정의하여 섹션이 생성되는 비즈니스 로직을 정의하였습니다.

jjeongdong commented on src/main/java/com/example/projectlxp/section/service/impl/SectionServiceImpl.java:
```diff
@@ -0,0 +1,60 @@
+
+        if (findedSection != null) {
+            throw new IllegalStateException("동일한 순서로 Section을 생성하고 있습니다.");
+        }
```
이 부분도 Course를 찾는 부분처럼


```java
sectionRepository.findByCourseIdAndOrderNo(courseId, orderNo)
    .ifPresent(section -> {
        throw new IllegalStateException("동일한 순서로 Section을 생성하고 있습니다.");
    });
```
이런식으로 바꿀 수 있을 것 같습니다.

JSeHoone replied:
헉 ! 감사합니다 ! 
말씀해주신 내용으로 수정하고 Commit 하겠습니다 !

JSeHoone committed:
refactor : ifPresent를 이용하여 throw처리 변경

- findByCourseIdAndOrderNo 메서드에서 ifPresent를 이용하여 IllegalStateException으로 반환되도록 리팩토링

JSeHoone committed:
style : 오타 수정

JSeHoone committed:
Merge branch 'develop' into feat/#11-create-section
